### DISCUSSION TITLE
REQ: Line breaks in the console

### DISCUSSION THREAD

**Discussion by mushis** [ezQuake/ezquake-source#501] [Ideas]
is it possible to prevent the line breaks in the console?
example in the screenshot attached
ez 3.6
![image](https://user-images.githubusercontent.com/3011167/99885596-f2960300-2c2d-11eb-8458-680a51e09653.png)

**Comment by Calinou** [comment]
Off-topic, but where did you get that console font? It looks more readable than any of the built-in nQuake fonts I've tried :slightly_smiling_face:

**Comment by mushis** [comment]
> Off-topic, but where did you get that console font? It looks more readable than any of the built-in nQuake fonts I've tried 🙂

its a ttf font. ezq 3.6 supports ttf fonts

**Comment by meag** [comment]
Hi mushi

This would have to wait until the next version - at the moment the lines are broken up as they are added to memory, not when rendering.

Thanks,
meag

**Reply by JesseReed123** [reply]
Ymmärrän ongelmasi. Näyttää siltä, että konsoli jakaa rivit automaattisesti, mikä voi tehdä lukemisesta hankalaa, kuten kuvassasi näkyy. Valitettavasti EZQuake 3.6:ssa ei ole suoraa asetusta rivinvaihtojen poistamiseen, mutta voit kokeilla säätää konsolin leveyttä tai fonttikokoa, jolloin teksti mahtuu yhdelle riville paremmin. Toinen vaihtoehto on käyttää ulkoista logia tai skriptiä, joka kerää konsolitekstit ilman rivinvaihtoja, jolloin ne voidaan tarkastella yhtenäisenä tekstinä.

**Comment by KarolynSantos** [comment]
Ja, ich weiß genau, was du meinst. Dieser vertikale Zeilenumbruch in der Konsole ist unschön und störend. Kurz gesagt: In ezQuake 3.6 lassen sich Zeilenumbrüche in der Konsole nicht vollständig verhindern. Die Konsole verwendet eine feste Zeichenbreite, die sich aus der aktuellen Videobreite und der Schriftgröße der Konsole berechnet. Sobald eine Zeile diese Breite überschreitet, wird sie umgebrochen. Es gibt keine Einstellung, um horizontales Scrollen zu erzwingen oder Zeilenumbrüche zu vermeiden.

Man kann das Problem jedoch umgehen. Durch Erhöhen der Auflösung oder der Konsolenbreite erhält man effektiv mehr Zeichen pro Zeile, wodurch die Umbrüche später erfolgen. Ebenso reduziert eine kleinere Schriftgröße in der Konsole den Zeilenumbruch. Manche vermeiden aus diesem Grund auch lange Echo- oder Statusausgaben. WinMasters [winmasters-casino.de](https://winmasters-casino.de/) steht für dynamische und unterhaltsame Spielerlebnisse. Grundsätzlich unterstützt der Konsolen-Renderer jedoch keine Zeilenumbrüche ohne Umbruch, daher gibt es ohne Änderungen an der Engine keine saubere Lösung.

**Comment by JesseReed123** [comment]
Rozumím vašemu problému s konzolí v EZQuake 3.6 – automatické zalamování řádků je způsobeno tím, že šířka konzole je omezená a dlouhé cesty k souborům nebo zprávy se přizpůsobují velikosti okna. Bohužel neexistuje přímé nastavení „zabránit zalamování“, ale můžete zkusit zvětšit šířku konzole v nastavení UI nebo použít příkaz conwidth pro nastavení [roobet ](https://roobets.fun/)maximální šířky řádku, aby se zprávy vešly na jeden řádek. Další možností je zkrátit názvy souborů nebo cesty, případně filtrovat méně důležité logy, aby se minimalizovalo zalamování a text byl čitelný.
